packages/itwinui-react/src/core/utils/functions/dev.ts

```diff
@@ -2,14 +2,22 @@
  * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
  * See LICENSE.md in the project root for license terms and full copyright notice.
  *--------------------------------------------------------------------------------------------*/
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
 const isJest = typeof (globalThis as any).jest !== 'undefined';
+const isMocha =
+  typeof (globalThis as any).beforeEach !== 'undefined' &&
+  `${(globalThis as any).beforeEach}`.replace(/\s/g, '') ===
+    'function(name,fn){suites[0].beforeEach(name,fn);}';
+const isVitest = (typeof globalThis as any).__vitest_index__ !== 'undefined';
```

Comment:
nit: Maybe good to include the link of where you took this idea from. (https://github.com/JoshuaKGoldberg/console-fail-test/blob/main/src/environments/vitest.ts?rgh-link-date=2024-01-30T19%3A47%3A38Z).

E.g.: `// See: <link>`

It would also be better if you use a permalink instead of the regular link.

Reply:
The threads in this PR do include the link, so it might be enough to link to this PR

Reply:
It would be hard to search for the repo/file link in the collapsed comments of this PR. So if you're including the PR link, then you can add the link to the repo/file in the PR description.

Reply:
I agree with updating PR description

Reply:
Unrelated note: these comments actually appear in out build output; it would be nice to strip them after build